src/libs/actions/IOU.js

```diff
@@ -237,14 +265,18 @@ function createSplitsAndOnyxData(participants, currentUserLogin, amount, comment
         : ReportUtils.getChatByParticipants(participantLogins);
     const groupChatReport = existingGroupChatReport || ReportUtils.buildOptimisticChatReport(participantLogins);
 
+    const amountInCents = Math.round(amount * 100);
+    const groupTransaction = TransactionUtils.buildOptimisticTransaction(amountInCents, currency);
```

Comment:
**question:** Should we have a group transaction when doing bill splits, or just the transactions on the 1:1 IOU reports? I might've just added this because `IOUTransactionID` became a required parameter of `ReportUtils.buildOptimisticIOUReportAction`. It might have been the case that the IOU action on the group report was getting `originalMessage.IOUTransactionID` set to a random transactionID of a transaction that actually doesn't exist anywhere.

Reply:
Looking [at Auth](https://github.com/Expensify/Auth/blob/9c9336353f5688ab85b1ef1e3caadf13a6e4b583/auth/command/CreateIOUSplit.cpp#L119) it seems like we put the split transactions on invisible report and the message is then added with the group chat report linking to [the transaction id](https://github.com/Expensify/Auth/blob/9c9336353f5688ab85b1ef1e3caadf13a6e4b583/auth/command/CreateIOUSplit.cpp#L133-L141).

I am not sure sure here, essentially I think we treated the group report action more like something in place but actionable reportActions are in the 1:1 chats. And also given the transaction is not linked to this report using the reportID, I dont think we would end up fetching them. 

We will need the transaction details to edit the request though so I think we will have to create it and and fetch the transaction details too. however not sure whats the best way

Reply:
I left an answer [here](https://github.com/Expensify/Expensify/issues/277227#issuecomment-1521375458), but I think we'll need the split transaction to display the request preview in the group chat as well.